packages/story-editor/src/app/story/storyProvider.js

```diff
@@ -57,33 +57,26 @@ function StoryProvider({ storyId, initialEdits, children }) {
   useEffect(() => setHashPageId(current), [current, setHashPageId]);
 
   // Generate current page info.
-  const {
-    currentPageId,
-    currentPageIndex,
-    currentPageNumber,
-    currentPage,
-    currentPageHash,
-  } = useMemo(() => {
-    if (!current) {
+  const { currentPageId, currentPageIndex, currentPageNumber, currentPage } =
+    useMemo(() => {
+      if (!current) {
+        return {
+          currentPageId: null,
+          currentPageIndex: null,
+          currentPageNumber: null,
+          currentPage: null,
+        };
+      }
+      const index = pages.findIndex(({ id }) => id === current);
+      const number = index + 1;
+      const page = pages[index];
       return {
-        currentPageId: null,
-        currentPageIndex: null,
-        currentPageNumber: null,
-        currentPage: null,
-        currentPageHash: null,
+        currentPageId: current,
+        currentPageIndex: index,
+        currentPageNumber: number,
+        currentPage: page,
```

Comment:
Didn't change anything here beyond removing the `currentPageHash` which we don't need anymore

Reply:
we could honestly remove this memoization call directly too and just assign the variable directly in the block scope of the component and it wouldn't change the behavior at all.

This is because the actual page would maintain the same reference as it does with the memoization between renders and the rest of the values are primitives. also without the `JSON.stringify()` none of the assignment calls here are expensive, so no need to limit them

Reply:
let's do that!